frontend/webapp/src/pages/screen-recorder.js

```diff
@@ -0,0 +1,87 @@
+import { useState, useRef } from "react";
+
+export default function ScreenRecorder() {
+  const [isRecording, setIsRecording] = useState(false);
+  const [videoUrl, setVideoUrl] = useState(null);
+  const mediaRecorder = useRef(null);
+  const chunks = useRef([]);
+
+  const startRecording = async () => {
+    try {
+      // Request screen recording permission
+      const stream = await navigator.mediaDevices.getDisplayMedia({
```

Comment:
even after the recording stops, the browser url window still has a 'recording' icon and shows "this tab is sharing your screen'

Reply:
i will try to solve it sir

Reply:
no this is not what we are expecting,
when we click on stop recording, it should stop sharing automatically, we should not need to do it manually.

Reply:
okay sir let me enhance the code